core/include/jiminy/core/stepper/AbstractStepper.h

```diff
@@ -0,0 +1,82 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+///
+/// \brief      Generic interface for a stepper to integrate the equation of dynamics.
+///
+/// \details    The generalized position of a mechanical system evolves not in a vector space,
+///             but in a Lie group: this is visible for instance in the fact that quaternions
+///             have four components, but only three are meaningful, as quaternion must remain
+///             unitary to represent a rotation. As such, the velocity vector v is not the
+///             term-by-term derivative of the configuration q.
+///             This means that classical stepper implementations cannot be used as is: this
+///             class thus defines an interface for implementing variable-step numerical
+///             solvers on this type of system.
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef JIMINY_ABSTRACT_STEPPER_H
+#define JIMINY_ABSTRACT_STEPPER_H
+
+#include <functional>
+
+#include "jiminy/core/Types.h"
+
+namespace jiminy
+{
+    class Robot;
+    struct systemState_t;
+
+    template<typename F1>
+    class AbstractStepper
```

Comment:
Sorry but I forgot that an abstract must NOT be templated (because the type is always unknown in advance in practice), so here we must hind the actual functor type using a std::function.